inspire_utils/name.py

```diff
@@ -246,6 +248,104 @@ def from_parts(
         name.title = title
         return ParsedName(name)
 
+    def generate_es_query(self, author_name, keyword="authors"):
```

Comment:
Why is this taking an author name? Don't you already know the name because you've passed it to `ParsedName`? I would also rename `keyword` to `field`, and document everything.